python/mxnet/io.py

```diff
@@ -306,32 +302,16 @@ def provide_data(self):
         if self.rename_data is None:
             return sum([i.provide_data for i in self.iters], [])
         else:
-            return sum([[(r[n], s) for n, s in i.provide_data] \
+            return sum([[DataDesc(r[x.name], x.shape, x.dtype) for x in i.provide_data] \
```

Comment:
It is a bit tricky here.
*x -> (name,shape) not (name, shape, dtype, layout) for current DataDesc class.